File: Python/FindDuplicateFileInSystem/FindDuplicateFileInSystem.py
class Solution(object):
	# (322ms)
    def findDuplicate(self, paths):
        """
        :type paths: List[str]
        :rtype: List[List[str]]
        """
        dt = {}
        for path in paths:
        	parts = path.split()
        	for part in parts[1:]:
        		subparts = part.split('(')
        		if subparts[1] not in dt:
        			dt[subparts[1]] = [parts[0] + '/' + subparts[0]]
        		else:
        			dt[subparts[1]].append(parts[0] + '/' + subparts[0])
        ans = []
        for value in dt.itervalues():
        	if len(value) > 1:
        		ans.append(value)
        return ans

File: JavaScript/FindDuplicateFileInSystem/FindDuplicateFileInSystem.js
/**
 * (345ms)
 * @param {string[]} paths
 * @return {string[][]}
 */
var findDuplicate = function(paths) {
    var dict = {};
    for(var path of paths) {
    	var parts = path.split(' ');
    	for(var i = 1; i < parts.length; i++) {
    		var subparts = parts[i].split('(');
    		if(dict.hasOwnProperty(subparts[1])) {
    			dict[subparts[1]].push(parts[0] + '/' + subparts[0]);
    		} else {
    			dict[subparts[1]] = [parts[0] + '/' + subparts[0]];
    		}
    	}
    }
    var ans = [];
    for(var value of Object.values(dict)) {
    	if(value.length > 1)
    		ans.push(value);
    }
    return ans;
};